### ISSUE TITLE
extractRequestParams: query params object required even when all fields are optional

### ISSUE THREAD

**Issue by Upgrade220** [acacode/swagger-typescript-api#1755]
## Describe the bug

When using `extractRequestParams: true`, if a route has only optional query parameters and no path parameters, the generated method signature incorrectly marks the combined params object as required:

```typescript
// generated — wrong
getItems(query: GetItemsParams, params?: RequestParams): Promise<...>
```

The caller is forced to pass an explicit argument even if they want to use defaults:
```typescript
// must write this
api.items.getItems({})

// but should be able to write this
api.items.getItems()
```

## Expected behavior

When all query fields are optional and there are no path params, the combined params object should have a default value of `{}`:

```typescript
// expected
getItems(query: GetItemsParams = {}, params?: RequestParams): Promise<...>
```

## Root cause

`procedure-call.ejs` hardcodes `optional: false` for the `requestParams` argument regardless of whether all query fields are optional. The `allFieldsAreOptional` flag is already computed correctly in the source (`requestParamsSchema.typeData.allFieldsAreOptional`) but is never surfaced to the template.

## Steps to reproduce

1. Define an OpenAPI route with only optional query parameters and no path parameters
2. Run generation with `--extract-request-params`
3. Observe the generated method signature — the params object is required
